specs/toolbar.md

```diff
@@ -0,0 +1,127 @@
+# Toolbar
+
+## Overview
+As defined by the W3C:
+
+> A toolbar is a container for grouping a set of controls, such as buttons, menubuttons, or checkboxes.
+> 
+> When a set of controls is visually presented as a group, the toolbar role can be used to communicate the presence and purpose of the grouping to screen reader users. Grouping controls into toolbars can also be an effective way of reducing the number of tab stops in the keyboard interface.
+
+Based on the above definition, it appears that as it currently stands, all elements within a toolbar _should_ be interactive controls. This includes support for [disabled controls](#disabled-controls).
+
+### Use Cases
+
+Used anywhere someone may want to visually and structurally group related interactive controls.
+
+- Tom is reading his favorite internet newspaper. Upon finishing an article, he notices that it has accrued nearly one thousand comments. Tom decides to read the comments and he ends up writing a response to someone he disagrees with. As he gets ready to submit, he realizes it will be easier for others to read his comment if he formats it. Tom sees a toolbar with formatting options and is able to bold all the words he deems important. He submits his comment and waits to see what happens next.
+
+### Prior Art/Examples
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/toolbar/definition)
+- [Material UI](https://material-ui.com/api/toolbar/)
+- [WAI-ARIA](https://w3c.github.io/aria-practices/examples/toolbar/toolbar.html)
+
+---
+
+### API
+
+*Component name:*
+- `fast-toolbar`
+
+*Attributes:*
+- `orientation`: an enum
+    - horizontal (default)
+    - vertical
+
+*Slots:*
+- `default`
+
+### Anatomy and Appearance
+
+*Template:*
+```html
+<template
+    role="toolbar"
+    orientation="${x => x.orientation}"
+>
+    <slot></slot>
+</template>
+```
+
+## Implementation
+
+**With visible label:**
+```html
+<h3 id="label">Toolbar visible label</h3>
+<fast-toolbar
+    orientation="horizontal" 
+    aria-labelledby="label"
+>
+    <div>
+        <fast-button>Bold</fast-button>
+        <fast-button>Italic</fast-button>
+        <fast-button>Underline</fast-button>
+    </div>
+    <select>
+        <option>Option 1</option>
+        <option>Option 2</option>
+        <option>Option 2</option>
+    </select>
+    <div>
+        <fast-button>More</fast-button>
+    </div>
+</fast-toolbar>
+```
+
+**With invisible label:**
+```html
+<fast-toolbar
+    orientation="horizontal" 
+    aria-label="Toolbar label"
+>
+    <div>
+        <fast-button>Bold</fast-button>
+        <fast-button>Italic</fast-button>
+        <fast-button>Underline</fast-button>
+    </div>
+    <select>
+        <option>Option 1</option>
+        <option>Option 2</option>
+        <option>Option 2</option>
+    </select>
+    <div>
+        <fast-button>More</fast-button>
+    </div>
+</fast-toolbar>
+```
```

Comment:
Do we need to support "groups" of elements within the toolbar to create visual subgroups within a toolbar but keep the children of the group as part of the toolbar's linear focus queue?
 ```
<Toolbar>
            <Button>Item 1</Button>
            <button>Item 2</button>
            <button>Item 3</button>
            <ToolbarItemGroup>
                <button>Item 4</button>
                <button>Item 5</button>
            </ToolbarItemGroup>
</Toolbar>
```

Reply:
It would be ideal if we don't have to and I think the initial intent should be to attempt without this. My intent with the divs shown in the example above was to specifically illustrate that the nesting structure within the toolbar shouldn't really matter.